src/python/gem5/components/boards/se_binary_workload.py

```diff
@@ -105,13 +106,17 @@ def set_se_binary_workload(
         process.executable = binary_path
         process.cmd = [binary_path] + arguments
         if stdin_file is not None:
-            process.input = stdin_file.get_local_path()
+            if isinstance(stdin_file, Path):
+                process.input = FileResource(stdin_file).get_local_path()
+            process.input = FileResource(Path(stdin_file)).get_local_path()
```

Comment:
I don't understand this.

The construction of this function still says `stdin_file` is a `Optional[FileResource]` defaulted to `None`, so if someone is passing `stdin_file` of type `Path` they are just wrong. 



I get the feeling here there's implication this is either a `str` or `Path` and this handles the conversion. Though in the case where `stdin_file` is a Path you're setting `process.input` on line 110 then immediately overriding it on line 111.

Reply:
@kunpai You still haven't addressed this.